include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,775 @@
+    bool vertexPipelineStoresAndAtomics = false;
+    bool fragmentStoresAndAtomics = false;
+    bool shaderTessellationAndGeometryPointSize = false;
+    bool shaderImageGatherExtended = false;
+    bool shaderStorageImageExtendedFormats = false;
+    bool shaderStorageImageMultisample = false;
+    bool shaderStorageImageReadWithoutFormat = false;
+    bool shaderStorageImageWriteWithoutFormat = false;
+    bool shaderUniformBufferArrayDynamicIndexing = false;
+    bool shaderSampledImageArrayDynamicIndexing = false;
+    bool shaderStorageBufferArrayDynamicIndexing = false;
+    bool shaderStorageImageArrayDynamicIndexing = false;
+    bool shaderClipDistance = false;
+    bool shaderCullDistance = false;
+    bool vertexAttributeDouble = false; // shaderFloat64
+    bool shaderInt64 = false;
+    bool shaderInt16 = false;
+    bool shaderResourceResidency = false;
+    bool shaderResourceMinLod = false;
+    
+    // [TODO] cause we haven't implemented sparse resources yet
+    //VkBool32    sparseBinding;
+    //VkBool32    sparseResidencyBuffer;
+    //VkBool32    sparseResidencyImage2D;
+    //VkBool32    sparseResidencyImage3D;
+    //VkBool32    sparseResidency2Samples;
+    //VkBool32    sparseResidency4Samples;
+    //VkBool32    sparseResidency8Samples;
+    //VkBool32    sparseResidency16Samples;
+    //VkBool32    sparseResidencyAliased;
+    
+    bool variableMultisampleRate = false;
+    bool inheritedQueries = false;
+
+    /* Vulkan 1.1 Core */
+    bool storageBuffer16BitAccess = false;
+    bool uniformAndStorageBuffer16BitAccess = false;
+    bool storagePushConstant16 = false;
+    bool storageInputOutput16 = false;
+    
+    // [TODO] do not expose multiview yet
+    //VkBool32           multiview;
+    //VkBool32           multiviewGeometryShader;
+    //VkBool32           multiviewTessellationShader;
+    
+    // [Future TODO]:
+    //VkBool32           variablePointersStorageBuffer;
+    //VkBool32           variablePointers;
+    
+    //VkBool32           protectedMemory; // [DO NOT EXPOSE] not gonna expose until we have a need to
+ 
+    // [DO NOT EXPOSE] Enables certain formats in Vulkan, we just enable them if available or else we need to make format support query functions in LogicalDevice as well
+    //VkBool32           samplerYcbcrConversion;
+    bool shaderDrawParameters = false;
+
+
+
+
+    /* Vulkan 1.2 Core */
+
+    bool samplerMirrorClampToEdge = false;          // ALIAS: VK_KHR_sampler_mirror_clamp_to_edge
+    bool drawIndirectCount = false;                 // ALIAS: VK_KHR_draw_indirect_count
+
+    // or VK_KHR_8bit_storage:
+    bool storageBuffer8BitAccess = false;
+    bool uniformAndStorageBuffer8BitAccess = false;
+    bool storagePushConstant8 = false;
+    
+    // or VK_KHR_shader_atomic_int64:
+    bool shaderBufferInt64Atomics = false;
+    bool shaderSharedInt64Atomics = false;
+   
+    // or VK_KHR_shader_float16_int8:
+    bool shaderFloat16 = false;
+    bool shaderInt8 = false;
```

Comment:
whatever doesn't need API changes or special GL or common (validation) backend code (so not `storagePushConstant8`), enable always and just report as a limit